src/transaction/mod.rs

```diff
@@ -131,19 +102,9 @@ impl DbcTransaction {
             return Err(Error::DbcIdNotUniqueAcrossInputs);
         }
 
-        // Check that the input and output blinded amounts are equal.
-        let input_sum: RistrettoPoint = self
-            .inputs
-            .iter()
-            .map(|i| i.blinded_amount)
-            .map(RistrettoPoint::from)
-            .sum();
-        let output_sum: RistrettoPoint = self
-            .outputs
-            .iter()
-            .map(BlindedOutput::blinded_amount)
-            .map(RistrettoPoint::from)
-            .sum();
+        // Check that the input and output amounts are equal.
+        let input_sum: u64 = self.inputs.iter().map(|i| i.amount.value).sum();
+        let output_sum: u64 = self.outputs.iter().map(|o| o.amount.value).sum();
 
```

Comment:
We might want to check for overflows here, on both inputs and outputs. 

```
input = 10
output1 = U64_MAX
output2 = 11

input == output1 + output2
```

In the above scenario, transaction is balanced!

Reply:
Just checked, RUST panics in that case! We're safe! But we might want to check so our program doesn't crash!

Reply:
Aha! Thanks for pointing that out, will use `checked_add` here!